Documentation/security/threat-model.rst

```diff
@@ -0,0 +1,751 @@
+| data             | settings could cause denial of service.           |
++------------------+---------------------------------------------------+
+
+Recommended Controls
+^^^^^^^^^^^^^^^^^^^^
+
+- `Kubernetes RBAC`_ should be configured to only grant necessary permissions
+  to users and service accounts. Access to resources in the ``kube-system``
+  and ``cilium`` namespaces in particular should be highly limited.
+- Kubernetes audit logs should be used to automatically review requests
+  made to the API server, and detections should be configured to
+  alert on suspicious activity.
+
+.. _Kubernetes RBAC: https://kubernetes.io/docs/reference/access-authn-authz/rbac/
+
+.. _kv-store-attacker:
+
+Cilium Key-value Store Attacker
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Cilium can use :ref:`an external key-value store <k8s_install_etcd>`
+such as etcd to store state. In this scenario, we consider a user with
+network access to the Cilium etcd endpoints and credentials to access
+those etcd endpoints. The credentials to the etcd endpoints are stored
+as Kubernetes secrets; any attacker would first have to compromise these
+secrets before gaining access to the key-value store.
+
+.. rst-class:: wrapped-table
+
++------------------+---------------------------------------------------+
+| **Threat         | **Identified STRIDE threats**                     |
+| surface**        |                                                   |
++==================+===================================================+
+| Cilium agent     | None                                              |
++------------------+---------------------------------------------------+
+| Cilium           | The ability to create or modify Identities or     |
+| configuration    | Endpoints in etcd would allow an attacker to      |
+|                  | “give” any pod any identity. The ability to spoof |
+|                  | identities in this manner might be used to        |
+|                  | escalate a single node compromise to a multi-node |
+|                  | compromise, for example by spoofing identities to |
+|                  | undermine ingress segmentation rules that would   |
+|                  | be applied on remote nodes.                       |
++------------------+---------------------------------------------------+
+| Cilium eBPF      | None                                              |
+| programs         |                                                   |
++------------------+---------------------------------------------------+
+| Network data     | None                                              |
++------------------+---------------------------------------------------+
+| Observability    | None                                              |
+| data             |                                                   |
++------------------+---------------------------------------------------+
+
+Recommended Controls
```

Comment:
I wonder if there's also some etcd authorization configuration that's worth noting.

Reply:
In our own docs the only suggestion we make is to [enable TLS](https://docs.cilium.io/en/stable/installation/k8s-install-external-etcd/#optional-configure-the-ssl-certificates) 

Is there anything in particular you were thinking of? I'm wary of recommending security controls that might not play nicely with Cilium if we haven't tested them.

Reply:
I don't think we have any guides, just generally wondering about the notion of RBAC with etcd. If we don't have any idea about that then we can leave this suggestion out for now.